Source/RapyutaSimulationPlugins/Private/Core/RRGameState.cpp

```diff
@@ -64,25 +67,53 @@ void ARRGameState::StartSim()
     UE_LOG(LogRapyutaCore, Display, TEXT("MAX SPLIT SCREEN PLAYERS: %d"), maxSplitscreenPlayers);
     verify(SCENE_INSTANCES_NUM <= maxSplitscreenPlayers);
 
+    // 0- Stream level & Fetch static-env actors
+    SetupEnvironment();
+
     for (int8 i = 0; i < SCENE_INSTANCES_NUM; ++i)
     {
-        // 0 - Create a new one added into [SceneInstanceList], each of which houses ~Common objects, and configures
+        // 1 - Create a new one added into [SceneInstanceList], each of which houses ~Common objects, and configures
         // ~CommonClass, ~SceneDirectorClass
         CreateSceneInstance(i);
 
-        // 1 - Create <SceneType>Common & <Plugin>Common objects
+        // 2 - Create <SceneType>Common & <Plugin>Common objects
         CreateServiceObjects(i);
 
-        // 2 - Trigger OnStartSim() for creating plugins' own common artifacts
+        // 3 - Trigger OnStartSim() for creating plugins' own common artifacts
         StartSubSim(i);
 
-        // 3 - Do preliminary configuration/spawning for the Sim operation
+        // 4 - Do preliminary configuration/spawning for the Sim operation
         InitializeSim(i);
 
         UE_LOG(LogRapyutaCore, Display, TEXT("[ARRGameState]:: SIM SCENE INSTANCE [%d] STARTED SUCCESSFULLY! "), i);
     }
 }
 
+void ARRGameState::SetupEnvironment()
+{
+    FetchEnvStaticActors();
+}
+
+void ARRGameState::FetchEnvStaticActors()

```

Comment:
What is the purpose of fetching wall, floor actors?

Reply:
We need walls, and floor to be available (fetched from the map) in order for the first spawned pallet to detect its collision-free locations, which are used during pallet placement randomization